ObjectiveCloudant/Operations/CDTCouchOperation.h

```diff
@@ -128,6 +128,27 @@ typedef NS_ENUM(NSInteger, CDTObjectiveCloudantErrors) {
  */
 - (void)callCompletionHandlerWithError:(nonnull NSError *)error;
 
+/**
+ URL path for this operation.
+
+ Should be overridden by subclasses to create appropriate path.
+ */
+@property (nonnull, nonatomic, strong) NSString *httpPath;
+
+/**
+ HTTP method for this operation.
+
+ Should be overridden by subclasses to provide appropriate method.
+ */
+@property (nonnull, nonatomic, strong) NSString *httpMethod;
+
+/**
+ Request body for this operation; return `nil` if no body (e.g., for GET).
+
+ Should be overridden by subclasses to provide request body data.
+ */
+@property (nullable, nonatomic, strong) NSData *httpRequestBody;
```

Comment:
I also don't think we should expose these properties outside of the framework, since its currently internal implementation. If we make it a protocol as we discussed however, it would make nice extension points for people to add new operations if they need to use a Couch/Cloudant feature eg Search, Views etc.